apps/web/src/components/templates/TestWorkflowModal.tsx

```diff
@@ -0,0 +1,111 @@
+import { INotificationTrigger } from '@novu/shared';
+import { JsonInput } from '@mantine/core';
+import { Button, Title, Modal } from '../../design-system';
+import { inputStyles } from '../../design-system/config/inputs.styles';
+import { useMutation } from 'react-query';
+import { testTrigger } from '../../api/templates';
+import { useContext, useState } from 'react';
+import { errorMessage, successMessage } from '../../utils/notifications';
+import * as Sentry from '@sentry/react';
+import { AuthContext } from '../../store/authContext';
+
+export function TestWorkflowModal({
+  isVisible,
+  onDismiss,
+  trigger,
+}: {
+  isVisible: boolean;
+  onDismiss: () => void;
+  trigger: INotificationTrigger;
+}) {
+  const { currentUser } = useContext(AuthContext);
+  const { mutateAsync: triggerTestEvent } = useMutation(testTrigger);
+
+  const subscriberVariables = [{ name: 'subscriberId' }, ...(trigger?.subscriberVariables || [])];
+
+  const toTrigger = `{ 
+    ${subscriberVariables
+      ?.map((variable) => {
+        return `"${variable.name}": "${
+          (currentUser && currentUser[variable.name === 'subscriberId' ? 'id' : variable.name]) ?? 'REPLACE_WITH_DATA'
+        }"`;
+      })
+      .join(',\n    ')}
+}`;
```

Comment:
Can we add component tests for this modal to make sure all those variables are generated properly? Also looks like this behaviour is duplicated from the curl section of the trigger. Maybe we can reuse the code for both?

Reply:
I feel that reusing will look more awkward than not. As in the trigger snippet it is one block of code with extra code lines (like curl address or the novu import for example) and here we also substitute part of the variables to their actual values. But I get what you mean.